wp-modules/editor/js/src/utils/test/getCustomCategories.ts

```diff
@@ -0,0 +1,89 @@
+import getCustomCategories from '../getCustomCategories';
+
+describe( 'getCustomCategories', () => {
+	it.each( [
+		[
+			[],
+			[
+				{
+					label: 'First Category',
+					value: 'first-category',
+				},
+				{
+					label: 'Second Category',
+					value: 'second-category',
+				},
+				{
+					label: 'Third Category',
+					value: 'third-category',
+				},
+			],
+			[],
+		],
+		[ [ 'third-category' ], [], [] ],
+		[
+			[ 'first-category', 'second-category', 'third-category' ],
+			[
+				{
+					label: 'First Category',
+					value: 'first-category',
+				},
+				{
+					label: 'Second Category',
+					value: 'second-category',
+				},
+				{
+					label: 'Third Category',
+					value: 'third-category',
+				},
+			],
+			[],
+		],
+		[
+			[ 'first-category', 'second-category', 'third-category' ],
+			[
+				{
+					label: 'First Category',
+					value: 'first-category',
+				},
+				{
+					label: 'Second Category',
+					value: 'second-category',
+				},
+				{
+					label: 'Third Category',
+					value: 'third-category',
+					pm_custom: true,
+				},
+			],
+			[ 'Third Category' ],
+		],
+		[
+			[ 'first-category', 'second-category', 'third-category' ],
+			[
+				{
+					label: 'First Category',
+					value: 'first-category',
+				},
+				{
+					label: 'Second Category',
+					value: 'second-category',
+					pm_custom: true,
+				},
+				{
+					label: 'Third Category',
+					value: 'third-category',
+					pm_custom: true,
+				},
+			],
+			[ 'Second Category', 'Third Category' ],
+		],
+	] )(
+		'should get the custom categories created by Pattern Manager',
```

Comment:
Nice test coverage, from `[]`, to matched categories that don't have `pm_custom`, to matched categories that do have `pm_custom`.